app/views/session_timeout/_ping.js.erb

```diff
@@ -26,12 +30,12 @@ function success(data) {
       time_cutoff = new Date().getTime() + warning,
       show_warning = time_timeout < time_cutoff;
 
-  if (show_warning & !el) {
-    cntnr.insertAdjacentHTML('afterbegin', warning_info);
+  if (show_warning & !modal.shown) {
```

Comment:
I also wondered about that, this has been in the codebase since I started working on the app! I think it may have been a typo, but the bitwise comparison does evaluate correctly. I can change it if you'd like.